test/id.test.js

```diff
@@ -0,0 +1,151 @@
+// Copyright IBM Corp. 2013,2019. All Rights Reserved.
+// Node module: loopback-datasource-juggler
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+// This test written in mocha+should.js
+'use strict';
+
+/* global getSchema:false, connectorCapabilities:false */
+const async = require('async');
+const bdd = require('./helpers/bdd-if');
+const should = require('./init.js');
+const uid = require('./helpers/uid-generator');
+
+let ds;
+const ValidationError = require('..').ValidationError;
+
+describe.only('id property', function() {
+  before(function(done) {
+    ds = getSchema();
+    done();
+  });
+
+  context('unspecified (defaults to generated:true)', function() {
+    context('with forceId:true (default)', function() {
+      let IdCheckA;
+      before(async () => {
+        IdCheckA = ds.define('IdCheckA', {
+          name: String,
+        });
+        await ds.automigrate(IdCheckA.modelName);
+        await IdCheckA.destroyAll();
+      });
+
+      it('should not require id value', async () => {
+        await IdCheckA.create({name: 'Pablo'}).should.not.be.rejected();
+      });
+
+      it('should not allow id value to be specified', async () => {
+        await IdCheckA.create({id: 10, name: 'Pablo'}).should.be.rejectedWith(/can\'t be set/);
+      });
+    });
+
+    context('with forceId:false', function() {
+      let IdCheckB;
+      before(async () => {
+        IdCheckB = ds.define('IdCheckB', {
+          name: String,
+        }, {forceId: false});
+        await ds.automigrate(IdCheckB.modelName);
+        await IdCheckB.destroyAll();
+      });
+
+      it('should not require id value', async () => {
+        await IdCheckB.create({name: 'Pablo'}).should.not.be.rejected();
+      });
+
+      it('should allow id value to be specified', async () => {
+        await IdCheckB.create({id: 10, name: 'Pablo'}).should.not.be.rejected();
+      });
+    });
+  });
+
+  context('generated:false (default)', function() {
+    context('with forceId:true (default)', function() {
+      let IdCheckC;
+      before(async () => {
+        IdCheckC = ds.define('IdCheckC', {
+          name: String,
```

Comment:
I think you are missing the definition if `id` property here? Otherwise it's not clear to me how is this test context (`generated:false`) different from the previous one (`unspecified (defaults to generated:true)`).